packages/embark-code-generator/src/index.js

```diff
@@ -80,6 +80,10 @@ class CodeGenerator {
       cb(this.getEmbarkJsProviderCode());
     });
 
+    this.events.setCommandHandler('code-generator:embarkjs:set-provider-code', (cb) => {
+      cb(this.getSetProviderCode());
+    });
```

Comment:
Not a necessity but this can be a wonderful one-liner:

```
this.events.setCommandHandler('code-generator:embarkjs:set-provider-code', cb => cb(this.getSetProviderCode());
```